csharp/src/Apache.Arrow/Arrays/BooleanArray.cs

```diff
@@ -153,17 +182,25 @@ private void CheckIndex(int index)
                 new[] { nullBitmapBuffer, valueBuffer }))
         { }
 
-        public BooleanArray(ArrayData data) 
+        public BooleanArray(ArrayData data)
             : base(data)
         {
             data.EnsureDataType(ArrowTypeId.Boolean);
         }
 
         public override void Accept(IArrowArrayVisitor visitor) => Accept(this, visitor);
 
+        [Obsolete("GetBoolean does not support null values. Use GetValue instead (which this method invokes internally).")]
```

Comment:
This is causing warnings in the build now because the tests are still using this method. Can we update the tests to either:

1. Use the new method or
2. If they are explicitly testing this method, then put a `#pragma warning disable` to disable the warning in the build.

Reply:
Fixed